.circleci/config.yml

```diff
@@ -0,0 +1,201 @@
+  save_cache: &save_cache
+    save_cache:
+      <<: *cache_key
+      paths:
+        - ~/work/.gradle
+
+  accept_licenses : &accept_licenses
+    run:
+      name: Accept licenses
+      command: yes | sdkmanager --licenses || true
+
+  ## Workspace
+
+  workspace: &workspace
+    ~/work
+  attach_debug_workspace: &attach_debug_workspace
+    attach_workspace:
+      at: *workspace
+  persist_debug_workspace: &persist_debug_workspace
+    persist_to_workspace:
+      root: *workspace
+      paths:
+        - share_app/build/outputs/androidTest-results
+        - share_app/build/outputs/apk
+        - share_app/build/outputs/code-coverage
+        - share_app/build/test-results
+  attach_firebase_workspace: &attach_firebase_workspace
+    attach_workspace:
+      at: *workspace
+  persist_firebase_workspace: &persist_firebase_workspace
+    persist_to_workspace:
+      root: *workspace
+      paths:
+        - firebase
+
+  ## Docker image configurations
+
+  android_config: &android_config
+    working_directory: *workspace
+    docker:
+      - image: circleci/android:api-26-alpha
+    environment:
+      # Least invasive change to resolve out-of-memory error: https://discuss.circleci.com/t/circle-ci-v2-and-android-memory-issues/11207/9
+      _JAVA_OPTIONS: "-Xmx1024m"
+  gcloud_config: &gcloud_config
+    working_directory: *workspace
+    docker:
+      - image: google/cloud-sdk:latest
+
+jobs:
+
+  ## Build debug APK and instrumented test APK
+
+  build_debug:
+    <<: *android_config
+    steps:
+        - checkout
+        - *restore_cache
+        - *accept_licenses
+        - run:
+            name: Download dependencies
+            command: |
+                sudo chmod +x gradlew
+                ./gradlew androidDependencies
+        - *save_cache
+        - run:
+            name: Run code quality checks
+            command: ./gradlew pmd checkstyle lint findbugs
+        - store_artifacts:
+            path: share_app/build/reports
+            destination: reports
+        - run:
+            name: Assemble debug and test builds
+            command: |
+                ./gradlew assembleDebug -PdisablePreDex
+                ./gradlew assembleDebugAndroidTest -PdisablePreDex
+        - *persist_debug_workspace
+
+  ## Run unit tests
+
+  test_unit:
+    <<: *android_config
+    steps:
+        - checkout
+        - *restore_cache
+        - *accept_licenses
+        - run:
+            name: Download dependencies
+            command: |
+                sudo chmod +x gradlew
+                ./gradlew androidDependencies
+        - *save_cache
+        - run:
+            name: Run unit tests
+            command: ./gradlew testDebugUnitTest
+        - *persist_debug_workspace
+        - store_artifacts:
+            path: share_app/build/reports
+            destination: reports
+        - store_test_results:
+            path: share_app/build/test-results
+
+  ## Run instrumented tests
+
+  # test_instrumented:
```

Comment:
This can be cleaned up and later when the need arises, the configuration can be added back